python/ray/serve/_private/deployment_state.py

```diff
@@ -1180,10 +1219,14 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
         existing_info = self._target_state.info
         if existing_info is not None:
             # Redeploying should not reset the deployment's start time.
-            deployment_info.start_time_ms = existing_info.start_time_ms
+            if not self._target_state.deleting:
```

Comment:
`if not self._target_state.deleting`, what the reason for adding this check? is this an existing bug?

Reply:
I'm not sure, I don't know how often this case will occur (probably not often?). 
This is specifically for when a deployment has been deleted (e.g. the deployment graph changed) and then redeployed (e.g. the deployment graph changed back). 

The biggest change I made here is on line 1211, so that if all configs are the same but the current target state is deleting, the deployment isn't blocked from redeploying. I ran into this problem when debugging some unit tests, not sure why this pops up in my PR specifically.